javascript/ql/src/semmle/javascript/dataflow/Nodes.qll

```diff
@@ -706,7 +706,8 @@ class ClassNode extends DataFlow::SourceNode {
     result = getAClassReference(t.continue()).getAnInstantiation()
     or
     t.start() and
-    result.(AnalyzedNode).getAValue() = getAbstractInstanceValue()
+    result.(AnalyzedNode).getAValue() = getAbstractInstanceValue() and
+    not result = any(DataFlow::ClassNode cls).getAReceiverNode()
```

Comment:
What is the motivation for this change?

Reply:
While investigating the call graph changes I noticed some of the new edges were spurious. This was the root cause but didn't surface without the new steps.